homeassistant/components/alexa_local_control.py

```diff
@@ -0,0 +1,573 @@
+        service = SERVICE_TURN_ON if result else SERVICE_TURN_OFF
+
+        # Construct what we need to send to the service
+        data = {ATTR_ENTITY_ID: entity_id}
+
+        if brightness is not None:
+            data[ATTR_BRIGHTNESS] = brightness
+
+        if entity.domain.lower() in config.off_maps_to_on_domains:
+            # Map the off command to on
+            service = SERVICE_TURN_ON
+
+            # Caching is required because things like scripts and scenes won't
+            # report as "off" to Alexa if an "off" command is received, because
+            # they'll map to "on". Thus, instead of reporting its actual
+            # status, we report what Alexa will want to see, which is the same
+            # as the actual requested command.
+            self.cached_states[entity_id] = (result, brightness)
+
+        # Perform the requested action
+        self.hass.services.call(core.DOMAIN, service, data, blocking=True)
+
+        json_response = [create_hue_success_response(entity_id, 'on', result)]
+
+        if brightness is not None:
+            json_response.append(
+                create_hue_success_response(entity_id, 'bri', brightness))
+
+        return self.json(json_response)
+
+
+def parse_hue_api_put_light_body(request_json, entity):
+    """Parse the body of a request to change the state of a light."""
+    if 'on' in request_json:
+        if not isinstance(request_json['on'], bool):
+            return None
+
+        if request_json['on']:
+            # Echo requested device be turned on
+            brightness = 100
+            report_brightness = False
+            result = True
+        else:
+            # Echo requested device be turned off
+            brightness = 0
+            report_brightness = False
+            result = False
+
+    if 'bri' in request_json:
+        # Make sure the entity actually supports brightness
+        entity_features = entity.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
+
+        if (entity_features & SUPPORT_BRIGHTNESS) == SUPPORT_BRIGHTNESS:
+            try:
+                # Clamp brightness from 0 to 255
+                brightness = max(0, min(int(request_json['bri']), 255))
+            except ValueError:
+                return None
+
+            report_brightness = True
+            result = True if brightness > 0 else False
+
+    return (result, brightness) if report_brightness else (result, None)
+
+
+def entity_to_json(entity, is_on=None, brightness=None):
+    """Convert an entity to its JSON representation."""
+    if is_on is None:
+        is_on = entity.state == "on"
+
+    if brightness is None:
+        brightness = 255 if is_on else 0
+
+    name = entity.attributes.get(
+        ATTR_ECHO_FRIENDLY_NAME, entity.attributes['friendly_name'])
+
+    return {
+        'state':
+        {
+            'on': is_on,
+            'bri': brightness,
+            'reachable': True
+        },
+        'type': 'Dimmable light',

```

Comment:
Still need to put the correct type in here depending on if the entity supports dim/volume change or only supports turn on/off (Hue has both kinds of bulbs, just need to figure out what the non-dimmable light type is)